packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/SuccessTokenLongShortPairFinancialProductLibrary.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./LongShortPairFinancialProductLibrary.sol";
+import "../../../../common/implementation/Lockable.sol";
+
+/**
+ * @title Success Token Long Short Pair Financial Product Library.
+ * @notice Adds settlement logic to create success token LSPs. A success token pays out a fixed amount of
+ * collateral as a floor, with the remaining amount functioning like an embedded option. The embedded
+ * option can use the payout logic of other financial product libraries, and resemble a covered call, a
+ * KPI option, a range bond, a capped yield dollar, or custom payout logic in a new financial library.
+ */
+contract SuccessTokenLongShortPairFinancialProductLibrary is LongShortPairFinancialProductLibrary, Lockable {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    struct SuccessTokenLongShortPairParameters {
+        uint256 strikePrice;
+        uint256 basePercentage;
+    }
+
+    mapping(address => SuccessTokenLongShortPairParameters) public longShortPairParameters;
+
+    /**
+     * @notice Enables any address to set the strike price for an associated LSP.
+     * @param longShortPair address of the LSP.
+     * @param strikePrice the strike price for the covered call for the associated LSP.
+     * @param basePercentage the base percentage of collateral per pair paid out to long tokens, expressed
+     * with 1e18 decimals. E.g., a 50% base percentage should be expressed 500000000000000000, or 0.5 with
+     * 1e18 decimals. The base percentage can not be set to 0.
+     * @dev Note: a) Any address can set the initial strike price b) A strike price cannot be 0.
+     * c) A strike price can only be set once to prevent the deployer from changing the strike after the fact.
+     * d) For safety, a strike price should be set before depositing any synthetic tokens in a liquidity pool.
+     * e) longShortPair must expose an expirationTimestamp method to validate it is correctly deployed.
+     */
+    function setLongShortPairParameters(
+        address longShortPair,
+        uint256 strikePrice,
+        uint256 basePercentage
+    ) public nonReentrant() {
+        require(ExpiringContractInterface(longShortPair).expirationTimestamp() != 0, "Invalid LSP address");
+        SuccessTokenLongShortPairParameters memory params = longShortPairParameters[longShortPair];
+        require(params.strikePrice == 0 && params.basePercentage == 0, "Parameters already set");
+        require(basePercentage != 0, "Base percentage cannot be set to 0");
```

Comment:
Should you also require `strikePrice != 0`? This way the previous `require` always reverts after the params are set. Also in your comments you say that the strike price cannot be 0

Reply:
Basically, repeat the logic of the `require` statement on line 66